### PR TITLE
[Fleet][Security Solution] Update Kibana System and Fleet Server index permissions to enable File Operations

### PR DESCRIPTION
**Pull request by kevinlog**
This addresses an issue where file permissions for `kibana_system` and `elastic/fleet-server` were not set up with the right permissions for file operations in Fleet and Security such as requesting Agent Diagnostics and file related Endpoint response actions.

Specifically, we need to grant permissions for the `.fleet-fileds` indices.

Before this change, we get permissions issues when requesting Agent diagnostics from the host

image

image

After this change, we are able to get the diagnostics file
image

In addition to the above changes, we needed to grant the right permissions to ensure that we can perform rollovers via lifecycle management.

More info in this ticket: https://github.com/elastic/kibana/issues/162760

### CODE REVIEW

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStoreTests.java [hunk 1]
```diff
@@ -753,7 +753,8 @@ public void testKibanaSystemRole() {
             ".fleet-enrollment-api-keys",
             ".fleet-policies",
             ".fleet-actions-results",
-            ".fleet-servers"
+            ".fleet-servers",
+            ".fleet-fileds"
```
Comment (joshdover) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStoreTests.java:757]: Just for my own knowledge - why did we name this `fileds`?

Comment (kevinlog) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStoreTests.java:757]: these are now datastreams, it was done to differentiate from the legacy `.fleet-file` which were regular indices.

**Comment by n1v0lg** [comment]
@kevinlog since these are system data streams, just a call out that data stream lifecycle management won't work unless you also add privileges to the data stream lifecycles internal user: https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java#L154

There is an example there already for `.fleet-actions-results` -- you will also need to add `.fleet-fileds` (or whatever name you ultimately land on) there, if this data stream is to be managed via data stream lifecycles (cc @andreidan for visibility).

**Comment by kevinlog** [comment]
@n1v0lg thank you for the heads up!  I made the changes you suggested (in addition to fixing some errors in our ilm policies for ESS) - https://github.com/elastic/elasticsearch/pull/98075/commits/19e98f3b882d3f7035449bda32c8c3026f3349cf

**Review by joshdover — APPROVED** [review]

**Review by n1v0lg — APPROVED** [review]
LGTM from an ES security perspective 👍 

Let's also get a +1 for @elastic/kibana-security

**Review by jeramysoucy — APPROVED** [review]
As `.fleet-fileds` is a system index owned by Kibana, we're good with this. Thanks!

**Pull request merged**